ros2_generate_interface_docs/ros2_generate_interface_docs/main.py

```diff
@@ -17,30 +17,57 @@
 import argparse
 import os
 import sys
+import time
 
 from ros2_generate_interface_docs import msg_utils
 from ros2_generate_interface_docs import utils
 
 from rosidl_runtime_py import get_message_interfaces
 
 
-# generic function takes op and its argument
-def generate_interfaces(generate_index, generate_doc, interfaces,
-                        html_dir, template, interface_type):
+def generate_interfaces(interfaces, html_dir, template, interface_type):
+    """
+    Generate the index and documentation for each message.
+
+    Parameters
+    ----------
+    interfaces: dict of {str : str[]}
+        dictionary with the interface package name associated with all the interface
+        for this package.
+    html_dir: str
+        path to the directory to save the generated documentation
+    template: str
+        template name
+    interface_type: str
+        type of the interface: msg, action or srv
+
+    """
     for package_name, names in sorted(interfaces.items(), key=lambda item: item[0]):
         package_directory = os.path.join(html_dir, package_name)
         interface_type_directory = os.path.join(package_directory, interface_type)
         os.makedirs(interface_type_directory, exist_ok=True)
-        generate_index(package_name,
-                       package_directory,
-                       interfaces[package_name])
+        utils.generate_index(package_name,
+                             package_directory,
+                             interfaces[package_name])
         for name in names:
```

Comment:
I'm sorry I didn't mention this earlier, but I'm losing track of what name refers to here when reading through the rest of this function. I think this would be easier to follow if you used, 'for interface_name in interface_names'